.github/workflows/scripts/comments.mjs

```diff
@@ -90,53 +89,106 @@ export async function commentDiff(
+  } else {
+    if (fileSizeData.length) {
+      // File sizes
+      const fileSizeTitle = '### File sizes'
+      const fileSizeRows = (await getReadableFileSizes(fileSizeData)).map(
+        (fileSize) => Object.values(fileSize)
+      )
 
-  const distPath = join(path, 'dist')
-  const packagePath = join(path, 'packages/govuk-frontend/dist/govuk')
-
-  // File sizes
-  const fileSizeTitle = '### File sizes'
-  const fileSizeRows = [
-    ...(await getFileSizes(join(distPath, '**/*.{css,js,mjs}'))),
-    ...(await getFileSizes(join(packagePath, '*.{css,js,mjs}')))
-  ]
-
-  const fileSizeHeaders = ['File', 'Size']
-  const fileSizeTable = renderTable(fileSizeHeaders, fileSizeRows)
-  const fileSizeText = [fileSizeTitle, fileSizeTable].join('\n')
-
-  // Module sizes
-  const modulesTitle = '### Modules'
-  const modulesRows = (await Promise.all(modulePaths.map(getStats))).map(
-    ([modulePath, moduleSize]) => {
-      const { base, dir, name } = parse(modulePath)
+      const fileSizeHeaders = ['File', 'Size', 'Percentage change']
+      const fileSizeTable = renderTable(fileSizeHeaders, fileSizeRows)
+      fileSizeText = [fileSizeTitle, fileSizeTable].join('\n')
+    } else {
+      fileSizeText = 'No changes to asset file sizes.'
+    }
 
-      const statsPath = `docs/stats/${dir}/${name}.html`
-      const statsURL = new URL(statsPath, reviewAppURL)
+    if (modulesData.length) {
+      // Module sizes
+      const modulesTitle = '### Modules'
+      // Create an empty array to pass to the reducer so that jsdoc doesn't thing
+      // that the default accumulator value is `never[]`
+      /** @type {Array<Array<string|number>>} */
+      const modulesRowsEmpty = []
+      const modulesRows = (await getReadableFileSizes(modulesData)).reduce(
+        (accumulator, module) => {
+          if (module.percentage) {
+            const { base, dir, name } = parse(module.path)
+            const statsPath = `docs/stats/${dir}/${name}.html`
+            const statsURL = new URL(statsPath, reviewAppURL)
+            const moduleLink = `[${base}](${statsURL})`
+            const rowAlreadyExists = accumulator?.find(
+              (row) => row[0] === moduleLink
+            )
+
+            if (rowAlreadyExists) {
+              const existingRowIndex = accumulator.findIndex(
+                (row) => row === rowAlreadyExists
+              )
+
+              if (module.type === 'bundled') {
+                accumulator[existingRowIndex].splice(1, 0, module.size)
+                accumulator[existingRowIndex].splice(2, 0, module.percentage)
+              } else {
+                accumulator[existingRowIndex].push(module.size)
+                accumulator[existingRowIndex].push(module.percentage)
+              }
+            } else {
+              accumulator.push([moduleLink, module.size, module.percentage])
+            }
+
+            return accumulator
+          }
+
+          throw new Error(
```

Comment:
Does this only exist to appease TypeScript?

If we _know_ module.percentage will exist, should we consider either updating the `FileSize` typedef to make it non-optional, or split out a separate type where it's non-optional?

Reply:
If we need to keep the check then for readability I'd suggest flipping the logic here and throwing the error from the top of the function if module.percentage is falsey / not set.

That'd allow you to remove a level of nesting and moves the error closer to the logic that triggers it.

Reply:
I'm gonna try the separate type solution. Because we only apply percentage at the comparison point I was hoping I could just amend the object and use the same typedef. I'm hoping I can get rid of this.